benchmark/common.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <random>
+
+#include "../core/test/make_sparse.h"
+
+template <typename T> struct GenerateSparse {
+  auto operator()(int length) {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 100);
+
+    auto a = make_sparse_variable<T>(length);
+    unsigned long long size(0);
+
+    /* Generate a random amount of sparse data for each point */
+    auto vals = a.template sparseValues<T>();
+    for (scipp::index i = 0; i < length; ++i) {
+      const auto l = dis(gen);
+      size += l;
+      vals[i] = scipp::core::sparse_container<T>(l, i);
+    }
+
+    return std::make_tuple(a, sizeof(T) * size);
+  }
+};
+
+template <int NameLen> struct GenerateWithDataItems {
+  auto operator()(const int itemCount = 5, const int length = 100) {
+    Dataset d;
+    for (auto i = 0; i < itemCount; ++i) {
+      d.setData(std::to_string(i) + std::string(NameLen, 'i'),
+                makeVariable<double>(
+                    {{Dim::X, length}, {Dim::Y, length}, {Dim::Z, length}}));
+    }
+    return std::make_tuple(d, sizeof(double) * itemCount * length);
```

Comment:
Shouldn't this be `length^3`? It is also inconsistent with the code comment you have further down:

```static void Args_Dataset_copy(benchmark::internal::Benchmark *b) {
  b->RangeMultiplier(2)
      ->Ranges({/* Item count */ {1, 16},
                /* Item length */ {32, 64}})
      ->Unit(benchmark::kMicrosecond);
}```

Reply:
Yes, it should be `length^3`. I'm not sure how this is inconsistent with the comment you mention though?

Reply:
"Axis length" is still not really useful. What the reader wants to know in a copy benchmark is how many elements or bytes are copied.

Reply:
I can either add the element count as an additional counter since it is already calculated or make the datasets 1D.

Trying to calculate axis lengths based on dimensionality and the number of elements you want isn't a good approach IMO (I tried this in the Variable benchmarks).

Reply:
Or rename to `GenerateDataset3D`, then `"axis length"` becomes more meaningful?